### PR TITLE
GEN-2358: RLV compatiblity

### PR DESCRIPTION
**Pull request by fedealconada**
## Motivation

Make sure that Periphery v2 is backwards compatible with RLVs.

## Solution

`AutoRoller.sol` calls the following functions of `Periphery.sol`:
- [`periphery.MIN_YT_SWAP_IN()`](https://github.com/sense-finance/sense-v1/blob/dev/pkg/core/src/Periphery.sol#L36)
- [[`periphery.swapYTsForTarget(address(adapter), maturity, excessBal);`](](https://github.com/sense-finance/sense-v1/blob/dev/pkg/core/src/Periphery.sol#L319)url)
- [`periphery.spaceFactory()`](https://github.com/sense-finance/sense-v1/blob/dev/pkg/core/src/Periphery.sol#L61)
- `periphery.sponsorSeries(address(adapter), _maturity, true);`

While the first 3 exist on the Periphery v2, `sponsorSeries` does not exist with the signature mentioned in the example above (it has been modified). Hence, similar to what we did for `swapYTsForTarget`, we need to create a new `sponsorSeries` function which keeps the compatibility.

## Implementer Checklist

- [ ]  [OPTIONAL] Create a reference implementation (in python or JS)
- [ ]  Create a short "Motivation" section for the PR
- [ ]  Write a spec for the feature and put it in the PR description – basic function names and expected state transitions is OK
- [ ]  Get the PR reviewed by at least two people

If this is your first time reviewing smart contract changes
- [ ]  Review the [Solcurity](https://github.com/Rari-Capital/solcurity) standard

If smart contract changes were made
- [ ]  Check all of the new revert paths with concrete tests
- [ ]  Check all of the new storage slot writes with concrete tests – do not make the passing test case the trivial case!
- [ ]  Go line-by-line through the new code and ensure it has all been covered by concrete tests – since we don’t have a coverage engine for foundry yet, you can imagine how one might check that each different code path is covered
- [ ]  Simplify the implementation and spend some time trying to minimize gas costs
- [ ]  Write fuzz tests for the feature – try everything to break the implementation (is this function monotonically in/decreasing, should it always be less than something else, etc)
- [ ]  Capture bugs discovered during the above steps in concrete tests (regression tests)
- [ ]  Add integration tests – fuzz or concrete tests that test how the new code affects the entire system, either locally or with a mainnet fork
- [ ]  [OPTIONAL] Integrate the feature into the deployment scripts
- [ ]  [OPTIONAL] Add new sanity checks to the deployment scripts

If your PR uses or interacts with prices of any kind
- [ ] Make sure forge is running the test on the latest block
- [ ] Compare the value returned with some other source (like Curve)
- [ ] Check the `updatedAt` (if available) value of the contract and confirm that the price we are sourcing is in fact not older than X
- [ ] Ensure oracle liveness from the oracle maintainer, either through a public status interface or through written confirmation from a developer.

### CODE REVIEW

**Comment by jparklev** [comment]
Assuming this should be compared to #358, so going to update the base for easier diffing

File: pkg/core/src/Periphery.sol [hunk 1]
```diff
@@ -137,6 +153,30 @@ contract Periphery is Trust, IERC3156FlashBorrower {
...
+    }
+
+    function _sponsorSeries(
+        address adapter,
+        uint256 maturity,
+        bool withPool,
+        address stake,
+        uint256 stakeSize
+    ) internal returns (address pt, address yt) {
+        ERC20(stake).transferFrom(msg.sender, address(this), stakeSize);
```
Comment (jparklev) [inline pkg/core/src/Periphery.sol]: Could this double transfer `stake` for when calling the permit `sponsorSeries`? (once from `_transferFrom(permit, address(quote.sellToken), quote.amount);` and again here?)

Maybe we can just move this line directly into the non-permit `sponsorSeries`?

**Review by jparklev — COMMENTED** [review]
Other than the comment on the double transfer, this approach seems OK to me. Looks like https://github.com/sense-finance/auto-roller/pull/38 is passing with this, which is a positive sign

Much appreciated @fedealconada for jumping on this, and good shout suggesting the test @stevenvaleri. After this is all done, let's do a quick RCA together to make note of any things we can learn dev-wise from the periph stuff. In the meantime, we found this just in time to fit it in with the other sherlock re-review items :)

And looks like we're still in the limit 🤞 (using the currently set 100 optimizer runs)
Screen Shot 2023-05-04 at 12 20 26 AM

File: pkg/core/src/Periphery.sol (hunk 1 above)
Comment (fedealconada) [reply pkg/core/src/Periphery.sol]: Totally! My bad, was in rush yesterday but good catch. Definitely was meant to be on the non-permit `sponsorSeries()` as you suggest :)

**Comment by fedealconada** [comment]
> Assuming this should be compared to #358, so going to update the base for easier diffing

yes!

**Comment by fedealconada** [comment]
> Other than the comment on the double transfer, this approach seems OK to me. Looks like [sense-finance/auto-roller#38](https://github.com/sense-finance/auto-roller/pull/38) is passing with this, which is a positive sign
> 
> Much appreciated @fedealconada for jumping on this, and good shout suggesting the test @stevenvaleri. After this is all done, let's do a quick RCA together to make note of any things we can learn dev-wise from the periph stuff. In the meantime, we found this just in time to fit it in with the other sherlock re-review items :)
> 

agree!

> And looks like we're still in the limit 🤞 (using the currently set 100 optimizer runs) Screen Shot 2023-05-04 at 12 20 26 AM

glad to see this!

**Review by stevenvaleri — APPROVED** [review]
Change looks good to me. We should add a test for the new sponsorSeries function. Since it shares the same logic as the other function with a test, it is non-blocking for merge, but we should add.

**Comment by fedealconada** [comment]
> Change looks good to me. We should add a test for the new sponsorSeries function. Since it shares the same logic as the other function with a test, it is non-blocking for merge, but we should add.

pushed!

**Pull request merged**